.github/scripts/Release.py

```diff
@@ -0,0 +1,550 @@
+
+    Returns:
+      Metadata in the format that the Zenodo API expects.
+    """
+    # Generate the DOI author list from the authors in the project metadata
+    # TODO: Check the metadata file is consistent, maybe elsewhere:
+    zenodo_creators = []
+    for author_tier in ['Major', 'Regular', 'Minor']:
+        for author in metadata['Authors'][author_tier]:
+            zenodo_creator = dict(name=author['Name'])
+            if 'Orcid' in author:
+                zenodo_creator['orcid'] = author['Orcid']
+            if 'Affiliations' in author and len(author['Affiliations']) > 0:
+                zenodo_creator['affiliation'] = ' and '.join(
+                    author['Affiliations'])
+            zenodo_creators.append(zenodo_creator)
+    # Render the description to HTML
+    rendered_description = github.render_markdown_raw(metadata['Description'])
+    # Construct Zenodo metadata
+    return dict(title=metadata['Name'],
+                version=metadata['Version'],
+                publication_date=metadata['PublicationDate'],
+                description=rendered_description,
+                creators=zenodo_creators,
+                related_identifiers=[
+                    dict(identifier=metadata['Homepage'],
+                         relation='isDocumentedBy',
+                         scheme='url',
+                         resource_type='publication-softwaredocumentation'),
+                    dict(identifier=('https://github.com/' +
+                                     metadata['GitHub']),
+                         relation='isSupplementTo',
+                         scheme='url',
+                         resource_type='software')
+                ],
+                language='eng',
+                communities=[dict(identifier='sxs')],
+                keywords=metadata['Keywords'],
+                license=metadata['License'],
+                upload_type='software',
+                access_right='open')
+
+
+def prepare(metadata: dict, version_name: str, metadata_file: str,
+            readme_file: str, zenodo: Zenodo, github: Github,
+            check_only: bool):
+    # Validate new version name
+    match_version_name = re.match(VERSION_PATTERN + '$', version_name)
+    if not match_version_name:
+        raise ValueError(
+            "Version name '{}' doesn't match pattern '{}'.".format(
+                version_name, VERSION_PATTERN))
+    publication_date = '{}-{}-{}'.format(*match_version_name.groups())
+
+    # Validate latest version ID
+    latest_version_id = metadata['ZenodoId']
+    try:
+        latest_version_id_on_zenodo = zenodo.get_latest_version_id(
+            record_id=latest_version_id)
+    except requests.exceptions.HTTPError as err:
+        raise requests.exceptions.HTTPError((
+            "Are you re-running the script over a repository that already has "
+            "the unpublished new version ID inserted into Metadata.yaml?"
```

Comment:
Is this suggesting that running over such a repository would cause the error, or that I should be running over such a repository to eliminate the error?

Reply:
reworded the error message (the former is true)